src/command_helpers.rs

```diff
@@ -301,14 +248,41 @@ pub(crate) fn command_add_output_file(
     }
 }
 
+#[cfg(feature = "parallel")]
+fn forward_available_stderr_as_warnings(
+    child_stderr_reader: &mut Option<BufReader<std::process::ChildStderr>>,
+) {
+    if let Some(child_stderr_reader) = child_stderr_reader.as_mut() {
+        if let Ok(available) = child_stderr_reader.fill_buf() {
+            let stdout = io::stdout();
+            let mut consumed = 0;
+            for line in available.split_inclusive(|&b| b == b'\n') {
+                // Only forward complete lines, leave the rest in the buffer.
+                if let Some((b'\n', line)) = line.split_last() {
```

Comment:
Isn't it more efficient to check the last byte of the buffer directly?

Reply:
`write_warning` doesn't expect the line to end with `\n`, so I'd have to trim it off anyway. Looking at the implementation for `split_last`, it doesn't seem that expensive.

Reply:
Thanks, I misunderstood the API of split_last